mozci/push.py

```diff
@@ -1487,12 +1508,88 @@ def identify_permanent_failures(self, retriggers_count=5, queue_prefix="") -> No
                 )
 
             to_retrigger = retriggers_count - len(retriggers) - scheduled_retriggers
-            if to_retrigger <= 0:
-                # TODO: Analyze the result of the retriggers to classify between backfills and permanent failures
+            if to_retrigger <= 1:
+                if not scheduled_retriggers:
+                    # All tasks have been retriggered, check the results
+                    self.backfill_new_test_failure(failure, retriggers)
                 continue
             logger.info(f"Starting {to_retrigger} retriggers for task {failure.label}")
             failure.retrigger(self, times=to_retrigger)
 
+    def backfill_new_test_failure(self, failure: Task, retriggers: list[Task]) -> None:
+        """Analyze the result of the test retriggers to classify between backfills and permanent failures."""
+        if any(
+            retrigger.state in ("unscheduled", "pending", "running")
+            for retrigger in retriggers
+        ):
+            logger.warning(
+                f"Some retrigger did not finished for task {failure.label}, skipping."
+            )
+            return
+
+        # Ensure logs have been parsed in Treeherder for the original failure and all of its retriggers
+        treeherder_client = TreeherderClientSource()
+        try:
+            failure_job = treeherder_client.get_job_from_task(failure)
+            treeherder_client.check_logs_parsed(failure_job["id"], branch=self.branch)
+        except JobUnavailable:
+            return
+
+        # Rely on the bug_suggestions data from Treeherder to determine the number of similar failures among retriggers
+        bug_suggestions = treeherder_client.get_bug_suggestions(failure_job["id"])
+        terms = {
+            suggestions["search"]: suggestions["bugs"]["open_recent"]
+            for suggestions in bug_suggestions
+            if suggestions["search"].startswith("TEST-UNEXPECTED-")
+        }
+        if not terms:
+            logger.warning(
+                f"No bug suggestion related to the failure {failure.label} could be fetched from Treeherder, skipping."
+            )
+            return
+        # Filter out failures that already have an intermittent bug opened
+        terms = {
+            term: open_recent
+            for term, open_recent in terms.items()
+            if not self.has_intermittent_bug(term, open_recent)
```

Comment:
Initially, this could be done for new failures (`not ... has_intermittent_bug`). But it is likely better to ignore if bugs or internal issues have been been filed in the past and instead:

- If it's a permanent failure: backfill with one task per push.
- If it's intermittent with >50% of tasks on the same push with a shared failure message: backfill with 6 tasks per push.